dev/release/rat_exclude_files.txt

```diff
@@ -18,6 +18,8 @@ cpp/src/arrow/util/random.h
 cpp/src/arrow/status.cc
 cpp/src/arrow/status.h
 cpp/src/arrow/vendored/*
+cpp/src/arrow/flight/sql/odbc/odbcabstraction/whereami.cc
+cpp/src/arrow/flight/sql/odbc/odbcabstraction/whereami.h
 cpp/build-support/asan_symbolize.py
```

Comment:
Note to myself: After this PR is merged, I will make comment on https://github.com/apache/arrow/pull/40939:

1. Is it ok for `whereami.h` and `whereami.cc` to be added to rat exclusion? The `whereami` library itself is MIT-licensed.  
2. Do file `whereami.h` and `whereami.cc` need to be moved to `cpp/src/arrow/vendored/*`?